### PR TITLE
implement ConnectorAPI

### PR DESCRIPTION
**Pull request by adufilie**
# What? Why?
Implement Connector API to formalize and replace Redux "duck" conventions.

Changes proposed in this pull request:
- Rewrite duck.js to Connector.ts
- Correct typings in ClinicalInformation feature.
- Relax tslint rules
- Normalize indentation in some files

# Checks
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [ ] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [GifGrabber](http://www.gifgrabber.com/).

# Notify reviewers
Read our [Pull request merging
policy](../CONTRIBUTING.md#pull-request-merging-policy). If you are part of the
cBioPortal organization, notify the approprate team (remove inappropriate):

@cBioPortal/frontend

If you are not part of the cBioPortal organization look at who worked on the
file before you. Please use `git blame <filename>` to determine that
and notify them here:

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -73,7 +73,6 @@
     "redux-module-builder": "0.0.12-0",
     "redux-seamless-immutable": "^0.3.0",
     "redux-thunk": "^2.1.0",
-    "render-if": "^0.1.1",
```
Comment (alisman) [inline package.json]: i just started using this library (render-if) in my PR!  is there a reason it was deleted besides not being used

File: src/pages/home/getHomePageData.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import {CancerStudy, TypeOfCancer} from "../../shared/api/CBioPortalAPI";
+
+export type HomePageData = {cancerTypes:TypeOfCancer[], studies:CancerStudy[]};
+
+function getHomePageData():Promise<HomePageData>
```
Comment (alisman) [inline src/pages/home/getHomePageData.ts]: We should go head and call this something more feature specific than homePageData.

File: src/pages/patientView/clinicalInformation/ClinicalInformationContainer.tsx [hunk 3]
```diff
@@ -1,35 +1,37 @@
...
+    setTab?: void;
+};
 
-export class ClinicalInformationContainerUnconnected extends React.Component<IClinicalInformationContainerProps, {}> {
+@Connector.decorator
+export default class ClinicalInformationContainer extends React.Component<IClinicalInformationContainerProps, {}> {
 
     componentDidMount() {
-        this.props.loadClinicalInformationTableData();
+		this.props.loadClinicalInformationTableData && this.props.loadClinicalInformationTableData();
```
Comment (alisman) [inline src/pages/patientView/clinicalInformation/ClinicalInformationContainer.tsx]: I haven't seen this before.  Is the idea that if that method doesn't exist, it doesn't get invoked?   Shouldn't the type checking insure that it's present?  Home comes it's optional in props interface?

File: package.json (hunk 1 above)
Comment (adufilie) [inline package.json]: See changes to src/shared/components/enhancedFixedDataTable/EnhancedFixedDataTable.jsx

File: src/pages/patientView/clinicalInformation/ClinicalInformationContainer.tsx (hunk 3 above)
Comment (adufilie) [inline src/pages/patientView/clinicalInformation/ClinicalInformationContainer.tsx]: problems occur elsewhere when the props make it required.

Comment (alisman) [inline src/pages/patientView/clinicalInformation/ClinicalInformationContainer.tsx]: don't quite understand.  lets discuss elsehwere.  hopefully we wouldn't have to do this everywheer we call a prop method?

**Pull request merged**